BLAST_loop.py

```diff
@@ -16,9 +16,11 @@
 from Bio.Blast.Applications import NcbiblastnCommandline
 #help(NcbiblastnCommandline)
 import os
+import sys
+directory = sys.argv[-1]	#first argument input after 'python BLAST_loop.py' in command line will be stored in variable 'directory'
```

Comment:
You need to check the length of `sys.argv`; if nothing is specified, then it will try to use `sys.argv[0]` which is the name of this script itself. That will create a `FileNotFoundError` instead of something more descriptive.